src/network/network-er-x-sfp.md

```diff
@@ -0,0 +1,332 @@
+
+`local-port` defines the port on the loopback address that will let you interact with babeld. Mainly used for debugging.
+
+`interface ethX`  
+Listen to babeld protocol on and peer with anything on this interface.
+
+```
+filter [name] action allow
+filter [name] if [if]
+filter [name] type redistribute
+```
+These 3 lines makeup a rule called [name]. This rules will allow redistribution of all IPs on the [if] interface.
+
+
+### Example
+
+```
+set protocols babeld local-port 999
+set protocols babeld denydefault true
+set protocols babeld denydefaultlocal true
+set protocols babeld interface l2tpeth0
+set protocols babeld interface eth1
+set protocols babeld interface eth2
+set protocols babeld interface eth3
+set protocols babeld interface eth4
+
+set protocols babeld filter eth1 action allow
+set protocols babeld filter eth1 if eth1
+set protocols babeld filter eth1 type redistribute
+
+set protocols babeld filter eth2 action allow
+set protocols babeld filter eth2 if eth2
+set protocols babeld filter eth2 type redistribute
+
+set protocols babeld filter eth3 action allow
+set protocols babeld filter eth3 if eth3
+set protocols babeld filter eth3 type redistribute
+
+
+set protocols babeld filter eth4 action allow
+set protocols babeld filter eth4 if eth4
+set protocols babeld filter eth4 type redistribute
+
+set protocols babeld filter l2tpeth0 action allow
+set protocols babeld filter l2tpeth0 if l2tpeth0
+set protocols babeld filter l2tpeth0 type redistribute
+```
+
+## Gateway Loop issues
+
+The paradox of the gateway and routing babeld
+
+- DHCP provides the default route to exit to the internet
+- Tunnel is established by using the default route
+- BABELD receives the IP address from the neighbour on the other side of the tunnel
+- The default route gets replaced with the EXIT NODE
+- All internet traffic is rerouted through the tunnel
+- Since the tunnel uses internet traffic it is also rerouted
+- Tunnel collapses  since its now feeding into itself
+
+
+### Workaround 1 - Static IP/Route
+
+- Define static IP on `ETH0` (WAN FACING PORT)
+- Remove default gateway
+- Create route out to the gateway for only the exit node
+
+```
+delete interfaces ethernet eth0 address dhcp
+set interfaces ethernet eth1 address 192.168.2.254/24    
+
+delete protocols static route 0.0.0.0/0
+set protocols static route 199.195.250.209/32 next-hop 192.168.2.1
+```
+
+### Workaround 2 - Route Tables
+
+Route tables allow the use of the default route (0.0.0.0) on the node to still point to the local internet connection. Babeld is instructed to place all the installed routes into a new routing table. A rule is installed to use this routing table for all packets coming in from interfaces we use babeld on. Then finally routes for the local node are added manually since babeld will not install them since they are local. Rules and routes must be installed in both the IPv4 and IPv6 stacks.
```

Comment:
```suggestion
Route tables allow the use of the default route (0.0.0.0) on the node to still point to the local Internet connection. Babeld is instructed to place all the installed routes into a new routing table. A rule is installed to use this routing table for all packets coming in from interfaces we use babeld on. Then finally routes for the local node are added manually since babeld will not install them since they are local. Rules and routes must be installed in both the IPv4 and IPv6 stacks.
```